### DISCUSSION TITLE
Automatische Sicherung von DPs mit der aktuellen Signatures?

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2722] [General]
Es werden plötzlich DPs mit gesichert in `/opt/iobroker/iobroker-data/jarvis/0` die sonst nicht dort gesichert wurden.
Wird da automatisch nach `signature` gesucht und alle zugehörige DPs mit gesichert?
![image](https://github.com/user-attachments/assets/d2cca67b-a8f2-4926-b419-7cfe0bb94459)
Und auch von denen, die nicht "orig" Jarvis-DPs sind, werden mehrere Varianten angelegt, bzw. immer nur angehängt. -> Speicher wird weniger.

**Comment by Zefau** [comment]
Ist vermutlich eines deiner Scripts? Oder backitup? `saveJarvisDPs` ist jedenfalls nicht von jarvis

**Comment by Erwin40** [comment]
Ist mir auch aufgefallen und ich nutze ausschließlich den Backitup Adapter

**Comment by mcuiobroker** [comment]
@zefau Es muss von jarvis ausgehen. Wenn ich die jarvis Instanz stoppe und dann mit `saveJarvisDPs` die DPs sichere (nur in vis.0) dann werden keine Dateien in dem Verzeichnis angelegt. Backitup wurde auch ausgeschaltet, um testen, ob es daher kommt.
-> Jarvis erstellt die Dateien, warum auch immer.

**Comment by Erwin40** [comment]
Kann es auch damit zusammenhängen, dass nach vielen kleinen Änderungen Jarvis beim Speichern immer länger braucht, bis sogar zum Timeout?
Das Einzige das mit da geholfen hat, war ein Raspi Restart. Jarvis Restart bringt nicht, und dauert auch recht lange bis die Verbindung wieder steht.

**Comment by Zefau** [comment]
Ich kann es mir nicht erklären. Habe bei mir auch einige ungewollte Gäste:

![grafik](https://github.com/user-attachments/assets/581f8c6e-9dc2-4032-92d2-a0169702ce68)

Habe diese 3 MQTT Dateien in allen Ordner meiner aktiven jarvis-Instanzen.
Der Code zum Backup ist recht simpel. Ich habe jetzt (mit der nächsten beta) noch eine zusätzliche Bestätigung eingebaut, dass es ein jarvis Datenpunkt ist.

Bitte löscht die Dateien dann manuell. Es sollten dann keine neuen Dateien mehr angelegt werden, sofern der Fehler von jarvis kam.

**Comment by Erwin40** [comment]
Suppi, aber welche kann ich denn gefahrlos löschen? :-)

![grafik](https://github.com/user-attachments/assets/fc6959a8-f1b8-4ae7-a64f-66695216b3b0)

**Reply by Zefau** [reply]
alle die nicht mit `jarvis.0` anfangen

**Comment by mcuiobroker** [comment]
Dies sind in der v3.2.0 nicht mehr aktiv
![image](https://github.com/user-attachments/assets/0b9f7864-e034-4409-b31e-52a2f4093afe)

**Comment by Erwin40** [comment]
Ok, gelöscht.
Und ist das gut so, das z.B. jarvis.0.devices.json ~9MB groß ist, mittlerweile?
Ich weiß, ich glaub du hast da jetzt überall ne Historisieung mit drin. Aber wieviele Versionen speicherst du dort, wenn man fragen darf?

**Reply by Zefau** [reply]
Sollten 25 drin sein.
